### PR TITLE
feat: plugin auto complete for dooray

### PR DESCRIPTION
**Pull request by jwlee1108**
### Please check if the PR fulfills these requirements
- [ ] It's the right issue type on the title
- [ ] When resolving a specific issue, it's referenced in the PR's title (e.g. `fix #xxx[,#xxx]`, where "xxx" is the issue number)
- [ ] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes/features)
- [ ] Docs have been added/updated (for bug fixes/features)
- [ ] It does not introduce a breaking change or has a description of the breaking change

### Description

---
Thank you for your contribution to TOAST UI product. 🎉 😘 ✨

### CODE REVIEW

File: plugins/auto-complete/src/index.ts [hunk 1]
```diff
@@ -0,0 +1,52 @@
+import type { PluginContext, PluginInfo } from '@toast-ui/editor';
+import { autoComplete } from '@/plugins/plugin';
+import { autoCompleteTrigger } from '@/plugins/trigger';
+import { AutoCompleteAction, PluginOptions } from '@t/index';
+import { USER_ACTIONS } from '@/constants';
+import { exitCode } from 'prosemirror-commands';
```
Comment (jajugoguma) [inline plugins/auto-complete/src/index.ts:6]: 사용되지 않는것 같습니다!

File: plugins/auto-complete/src/plugins/plugin.ts [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+        const isCancelWithSpace = triggerType?.cancelWithSpace ?? true;
+
+        if (isCancelWithSpace && filter.length === 0 && (ev.key === ' ' || ev.key === 'Spacebar')) {
+          closeAutoComplete(view);
+          view.dispatch(view.state.tr.insertText(' '));
+          return true;
+        }
+
+        if (filter.length === 0 && ev.key === 'Backspace') {
+          context.pmRules.undoInputRule(view.state, view.dispatch);
```
Comment (jajugoguma) [inline plugins/auto-complete/src/plugins/plugin.ts:141]: #2451 머지하고 리베이스 한번 해줘야될 것 같습니다!

**Review by jajugoguma — APPROVED** [review]
리뷰 완료합니다!

한글 입력 시 `handleKeyDown`이 호출되지 않는 건 ProseMirror 내부 코드 한번 확인해봐야할 것 같습니다... ㅠㅠ

File: plugins/auto-complete/src/index.ts (hunk 1 above)
Comment (jwlee1108) [reply plugins/auto-complete/src/index.ts:6]: 수정할게요!

File: plugins/auto-complete/src/plugins/plugin.ts (hunk 2 above)
Comment (jwlee1108) [reply plugins/auto-complete/src/plugins/plugin.ts:141]: 네

**Pull request closed without merging**